tests/Doctrine/Tests/DBAL/Sharding/PoolingShardManagerTest.php

```diff
@@ -47,6 +47,10 @@ public function testSelectGlobal() : void
     {
         $conn = $this->createConnectionMock();
         $conn->expects($this->once())->method('connect')->with($this->equalTo(0));
+        $conn->method('getParams')
+            ->willReturn([
+                'shardChoser' => $this->createMock(ShardChoser::class),
```

Comment:
This illustrates one more advantage of using classes for data structures instead of associative arrays. Unlike the array element, a method return value would be automatically mocked.